File: c++/Human.cpp
// Saya Muhammad Alam Basalamah [2101677] mengerjakan Tugas dari Praktikum yang ke 2 dalam mata kuliah DPBO
// untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikan. Aamiin.

//pemanggilan library string dan default c++
#include <iostream>
#include <string>

using namespace std;//mengindikasi bahwa saya menggunakan semua fungsi kelas dan objek terdefinisi di namespace std dalam program ini

class Human{//pembuatan class Human sebagai parent / superclass
    private://membuat property private
        string nik;
        string nama;
        string jenis_kelamin;
    
    public://membuat property public
        Human(string nik, string nama, string jenis_kelamin){//membuat konstruktor
            //set nik, nama , jenis kelamin
            this->nik = nik;
            this->nama = nama;
            this->jenis_kelamin = jenis_kelamin;
        }

        //membuat getter
        string getNik(){
            return this->nik;
        }

        string getNama(){
            return this->nama;
        }

        string getGender(){
            return this->jenis_kelamin;
        }
};

class SivitasAkademik : public Human{//membuat class (inheritance) sebagai sub class
    private ://membuat property private
        string asal_universitas;
        string email_edu;
    
    public ://membuat property public
        SivitasAkademik(string nik, string nama, string jenis_kelamin, string asal_universitas, string email_edu )//membuat konstruktor
        : Human(nik, nama, jenis_kelamin){//memanggil konstruktor Human / parent untuk mengisi variabel variabel didalamnya
            //set asal univ, dan email
            this->asal_universitas = asal_universitas;
            this->email_edu = email_edu;
        }

        //membuat getter

        string getAsalUniversitas(){
            return this->asal_universitas;
        }

        string getEmail(){
            return this->email_edu;
        }
};

class Mahasiswa : public SivitasAkademik{//membuat class (inheritance) sebagai sub class
    private://membuat property private
        string nim;
        string fakultas;
        string prodi;

    public://membuat property public
        Mahasiswa(string nik, string nama, string jenis_kelamin, string asal_universitas, string email_edu, string nim, string prodi, string fakultas)//membuat konstruktor
        : SivitasAkademik( nik,  nama,  jenis_kelamin, asal_universitas, email_edu){//memanggil konstruktor Sivitas Akademik / parentnya untuk mengisi variabel yang ada didalamnya
            // set nim, fakultas, dan prodi
            this->nim = nim;
            this->fakultas = fakultas;
            this->prodi = prodi;
        }

        // membuat getter
        
        string getNim(){
            return this->nim;
        }

        string getFakultas(){
            return this->fakultas;
        }

        string getProdi(){
            return this->prodi;
        }
};

File: python/Human.py
# Saya Muhammad Alam Basalamah [2101677] mengerjakan Tugas dari Praktikum yang ke 2 dalam mata kuliah DPBO
# untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikan. Aamiin.

# pembuatan class Human sebagai superclass
class Human:
    # deklarai property private
    __nik = ""
    __nama = ""
    __jenis_kelamin = ""

    # membuat konstruktor
    def __init__(self, nik, nama, jenis_kelamin):
    # set nik, nama jenis kelamin
        self.__nik = nik
        self.__nama = nama
        self.__jenis_kelamin = jenis_kelamin
    
    # membuat getter
    def getNik(self):
        return self.__nik

    def getNama(self):
        return self.__nama
    
    def getGender(self):
        return self.__jenis_kelamin

# inheritance (pembuatan sub class)
class SivitasAkademik(Human):
    # deklarasi property private
    __asal_universitas = ""
    __email_edu = ""

    # membuat konstruktor
    def __init__(self, nik, nama, jenis_kelamin, asal_universitas, email_edu):
        # memanggil konstruktor dari kelas parent/superclass
        super().__init__(nik, nama, jenis_kelamin)
        # set asal univ, dan email
        self.__asal_universitas = asal_universitas
        self.__email_edu = email_edu
    
    # membuat getter
    def getAsalUniversitas(self):
        return self.__asal_universitas
    
    def getEmail(self):
        return self.__email_edu
    
# inheritance (pembuatan sub class)
class Mahasiswa(SivitasAkademik):
    # deklarasi property private
    __nim = ""
    __prodi = ""
    __fakultas = ""

    # membuat konstruktor
    def __init__(self, nik, nama, jenis_kelamin, asal_universitas, email_edu, nim, prodi, fakultas):
        # memanggil konstruktor dari kelas parentnya
        super().__init__(nik, nama, jenis_kelamin, asal_universitas, email_edu)
        # set nim, prodi, dan fakultas
        self.__nim = nim
        self.__prodi = prodi
        self.__fakultas = fakultas
    
    def getNim(self):
        return self.__nim
    
    def getProdi(self):
        return self.__prodi
    
    def getFakultas(self):
        return self.__fakultas